os/tools/mkbinheader.py

```diff
@@ -124,198 +185,212 @@ def get_static_ram_size(bin_type):
-    if check_optimize_config(cfg_path) == True:
-        binary_ram_size = int(dynamic_ram_size)
-    else:
-        binary_ram_size = int(static_ram_size) + int(dynamic_ram_size)
-        binary_ram_size = roundup_power_two(binary_ram_size)
-
-    if float(kernel_ver) <= 0.0 :
-        kernel_ver = 3.0
-    else :
-        kernel_ver = float(kernel_ver)
-
-    # based on comp_enabled, check if we need to compress binary.
-    # If yes, assign to bin_comp value for compression algorithm to use.
-    # Else, assign 0 to bin_comp to represent no compression
-    if 0 < int(comp_enabled) <= COMP_MAX :
-        bin_comp = int(comp_enabled)
-    else :
-        bin_comp = 0
-
-    # Compress data according to Compression Algorithm represented by bin_comp
-    # Run mkcompressimg tool with provided options. Read output compressed file into data.
-    if bin_comp > COMP_NONE :
-        fp_tmp = open("tmp", 'wb+')
-        fp_tmp.write(data)
-        fp_tmp.close()
-        os.system(mkbinheader_path + '/compression/mkcompressimg ' + comp_blk_size + ' ' + comp_enabled + ' tmp' + ' tmp_comp')
-        fp_tmp = open("tmp_comp", 'rb')
-        data = fp_tmp.read()
-        file_size = fp_tmp.tell()
-        fp_tmp.close()
-        os.system('rm tmp tmp_comp')
-
-    fp = open(file_path, 'wb')
-
-    fp.write(struct.pack('H', header_size))
-    fp.write(struct.pack('B', bin_type))
-    fp.write(struct.pack('B', bin_comp))
-    fp.write(struct.pack('B', main_priority))
-    fp.write(struct.pack('B', loading_priority))
-    fp.write(struct.pack('I', file_size))
-    fp.write('{:{}{}.{}}'.format(binary_name, '<', SIZE_OF_BINNAME, SIZE_OF_BINNAME - 1).replace(' ','\0'))
-    fp.write(struct.pack('I', int(binary_ver)))
-    fp.write(struct.pack('I', binary_ram_size))
-    fp.write(struct.pack('I', int(main_stack_size)))
-    fp.write(struct.pack('f', kernel_ver))
-
-    # parsing _vector_start address from elf information.
-    # _vector_start is only for ARM architecture. so it
-    # operates in the ARM architecture.
-    if bin_type == BIN :
-        os.system('readelf -s ' + elf_path_for_bin_type + ' | grep _vector_start > addr_file')
-        addr_fp = open("addr_file", 'rb')
-        jump_addr = addr_fp.read()
-
-        addr_fp.close()
-        os.remove('addr_file')
-
-        addr_data = jump_addr.split(' ')
-        addr_s = '0x%s' %addr_data[3]
-        addr = int(addr_s, 0)
-    else :
-        addr = 0
+    config_path = CONFIG_DIR + '/.config'
+    priority = get_config_value(config_path, "CONFIG_BM_PRIORITY_MAX=")
+    if priority > 0 :
+        BM_PRIORITY_MAX = priority
+    priority = get_config_value(config_path, "CONFIG_BM_PRIORITY_MIN=")
+    if priority > 0 :
+        BM_PRIORITY_MIN = priority
+
+    with open(file_path, 'rb') as fp:
+        # binary data copy to 'data'
+        data = fp.read()
+        file_size = fp.tell()
+        fp.close()
+
+        if binary_format == 'bin' or binary_format == 'BIN' :
```

Comment:
If we separate the kernel case, is there `bin` type for user binary?
If not, we can remove the code which is related to bin type.